activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -41,6 +41,10 @@ class AbstractAdapter
 
       attr_accessor :pool
       attr_reader :visitor, :owner, :logger, :lock
+
+      cattr_accessor :additional_type_records_cache, default: {}
```

Comment:
This is only used in the postgresql adapter, why it is being defined in the abstract adapter?